src/main/scala/beam/utils/osm/WayFixer.scala

```diff
@@ -0,0 +1,136 @@
+package beam.utils.osm
+
+import com.conveyal.osmlib.Way
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object WayFixer extends LazyLogging {
+  val HIGHWAY_TAG: String = "highway"
+  val LANES_TAG: String = "lanes"
+  val MAXSPEED_TAG: String = "maxspeed"
+
+  def fix(ways: java.util.Map[java.lang.Long, Way]): Unit = {
+    val nFixedHighways = ways.asScala.count {
```

Comment:
This utilizes some constructs oddly (exists and counts with mutations) - making it hard to understand the code. How about this:

    val (nFixedHighways, nFixedLanes) = ways.asScala.foldLeft((0,0)){ 
      case ((fixedHighwayCount, fixedLaneCount), (osmId, way)) =>
        (fixedHighwayCount + fixAndGetIncrementer(getFixedHighwayType(osmId, way), HIGHWAY_TAG), 
          fixedLaneCount + fixAndGetIncrementer(getFixedLane(osmId, way), LANES_TAG))
    }

...

    def fixAndGetIncrementer(toBeFixedOption: Option[String], tag: String): Int = {
      toBeFixedOption match {
          case Some(toBeFixed) => 
            replace(osmId, way, tag, toBeFixed)
            1
          case None => 0
        }
    }

*Note I changed the name to be `getFixedLane` as it doesn't seem to return plural